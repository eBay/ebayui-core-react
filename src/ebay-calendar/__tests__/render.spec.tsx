import React from 'react'
import { render, screen } from '@testing-library/react'
import { composeStory } from '@storybook/react'
import Meta, { Default, Navigable, RangeSelected } from './index.stories'

jest
    .useFakeTimers()
    .setSystemTime(new Date('2024-03-05').getTime())

const DefaultStory = composeStory(Default, Meta)
const NavigableStory = composeStory(Navigable, Meta)
const RangeSelectedStory = composeStory(RangeSelected, Meta)

describe('ebay-calendar rendering', () => {
    it('renders default story correctly', () => {
        const { container } = render(<DefaultStory />)
        expect(container.querySelector('.calendar')).toBeInTheDocument()
        expect(container.querySelector('.calendar__body')).toBeInTheDocument()
        expect(container.querySelector('.calendar__cell--current').textContent.length).toBeGreaterThan(0)
        expect(screen.getByRole('table')).toBeInTheDocument()
        expect(screen.getByText('Sun')).toBeInTheDocument()
        expect(screen.getByText('Mon')).toBeInTheDocument()
        expect(screen.getByText('Tue')).toBeInTheDocument()
        expect(screen.getByText('Wed')).toBeInTheDocument()
        expect(screen.getByText('Thu')).toBeInTheDocument()
        expect(screen.getByText('Fri')).toBeInTheDocument()
        expect(screen.getByText('Sat')).toBeInTheDocument()
    })

    it('renders navigable story correctly', async () => {
        const { container } = render(<NavigableStory />)
        expect(container.querySelector('.calendar')).toBeInTheDocument()

        const buttonPrev = screen.getByLabelText('Show February 2024')
        expect(buttonPrev).toHaveClass('icon-btn icon-btn--small icon-btn--transparent')
        expect(buttonPrev).toHaveAttribute('type', 'button')
        const svgPrev = buttonPrev.querySelector('svg')
        expect(svgPrev).toHaveClass('icon icon--chevron-left-24')
        expect(svgPrev).toHaveAttribute('aria-hidden', 'true')

        const buttonNext = screen.getByLabelText('Show May 2024')
        expect(buttonNext).toHaveClass('icon-btn icon-btn--small icon-btn--transparent')
        expect(buttonNext).toHaveAttribute('type', 'button')
        const svgNext = buttonNext.querySelector('svg')
        expect(svgNext).toHaveClass('icon icon--chevron-right-24')
        expect(svgNext).toHaveAttribute('aria-hidden', 'true')

        const [month1, month2] = screen.getAllByRole('heading', { level: 3 })
        expect(month1.textContent).toEqual('March 2024')
        expect(month2.textContent).toEqual('April 2024')

        expect(screen.getByRole('table', { name: 'March 2024' })).toBeInTheDocument()
        expect(screen.getByRole('table', { name: 'April 2024' })).toBeInTheDocument()

        // todo: fix it in code
        // const selectedDate = screen.getByLabelText('3-selected');

        // await waitFor(() =>
        // expect(selectedDate).toBeInTheDocument()
        // expect(selectedDate).toHaveAttribute('data-iso', '2024-01-03')
        // , { timeout: 5000 })
    })

    // todo: finish after bug fix
    it.skip('renders range selected story correctly', () => {
        const { container } = render(<RangeSelectedStory />)
        expect(container.querySelector('.calendar')).toBeInTheDocument()
        expect(screen.getByRole('table')).toBeInTheDocument()
    })
})
