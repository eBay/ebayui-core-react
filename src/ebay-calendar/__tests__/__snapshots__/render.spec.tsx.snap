// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ebay-calendar rendering renders navigable story correctly 1`] = `
<svg
  aria-hidden="true"
  class="icon icon--24"
  focusable="false"
  xmlns="http://www.w3.org/2000/svg"
>
  <use
    xlink:href="#icon-chevron-left-24"
  />
</svg>
`;

exports[`ebay-calendar rendering renders navigable story correctly 2`] = `
<svg
  aria-hidden="true"
  class="icon icon--24"
  focusable="false"
  xmlns="http://www.w3.org/2000/svg"
>
  <use
    xlink:href="#icon-chevron-right-24"
  />
</svg>
`;
